lib/cfn-nag/custom_rules/EC2NetworkAclEntryPortRangeRule.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class EC2NetworkAclEntryPortRangeRule < BaseRule
+  def rule_text
+    'EC2 NetworkACL Entry Port Range must be defined for TCP/UDP protocols and not allow all ports'
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F80'
+  end
+
+  def audit_impl(cfn_model)
+    violating_network_acl_entries = cfn_model.resources_by_type('AWS::EC2::NetworkAclEntry')
+                                             .select do |network_acl_entry|
+      violating_network_acl_entries?(network_acl_entry)
+    end
+
+    violating_network_acl_entries.map(&:logical_resource_id)
+  end
+
+  private
+
+  # Port Range is required for protocols "6" (TCP) and "17" (UDP)
+  def tcp_or_udp_protocol?(network_acl_entry)
+    %w[6 17].include?(network_acl_entry.protocol)
+  end
+
+  def port_range_params_exist?(network_acl_entry)
+    network_acl_entry.portRange.nil? ||
+      network_acl_entry.portRange['From'].nil? || network_acl_entry.portRange['To'].nil?
+  end
+
+  def full_port_range?(network_acl_entry)
+    (network_acl_entry.portRange['From'] == '0' && network_acl_entry.portRange['To'] == '65535') ||
+      (network_acl_entry.portRange['From'] == '-1' && network_acl_entry.portRange['To'] == '-1')
```

Comment:
is this an and?  i would confirm by actual converging a template...... might just be either one of from/to being -1 makes it all ports

Reply:
Thanks for pointing this out, helped me catch a glaring error and my fault for not testing it more thoroughly.  So it turns out that you cannot have BOTH the `From` and `To` fields equal the value of '-1', that will fail to build properly as an invalid port range.  You also must have a `From` field so in reality we can ignore the test of `To = '-1'` because that realistically cannot happen (I don't know if we want to concern ourselves with testing if either of these scenarios happens or not).  So the real case here is this, a fully open port range is:
1.  Opening all ports from 0 to 65535
OR
2. Setting `From` to equal '-1' which will set the field in the NACL to be essentially `ALL`

This also continues to be awkward since it states this uses an integer but is always single quoted within Cloudformation templates.

For now, I'm adjusting this match the case I mentioned here.

Reply:
Following up on this and through a lot of testing, I've found that there's no scenario for the NACLEntry PortRange that allows the usage of '-1' as either From or To within Cloudformation, it fails every time.  So I won't be testing for that.